metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -583,76 +583,76 @@ void optimize_anchors_in_row_diff(const std::string &graph_fname,
 
     logger->trace("Optimizing anchors");
 
-    std::vector<std::string> filenames;
+    std::vector<sdsl::int_vector_buffer<>> row_reduction;
     for (const auto &p : std::filesystem::directory_iterator(dest_dir)) {
         auto path = p.path();
         if (utils::ends_with(path, row_reduction_extension)) {
             logger->info("Found row reduction vector {}", path);
-            filenames.push_back(path);
+            row_reduction.emplace_back(path, std::ios::in, 1024 * 1024);
+
+            if (row_reduction.back().size() != row_reduction.front().size()) {
+                logger->error("Row reduction vectors have different sizes");
+                exit(1);
+            }
+            if (row_reduction.back().width() != row_reduction.front().width()) {
+                logger->error("Row reduction vectors have different width: {} != {}",
+                              row_reduction.back().width(),
+                              row_reduction.front().width());
+                exit(1);
+            }
         }
     }
 
-    if (!filenames.size()) {
+    if (!row_reduction.size()) {
```

Comment:
I still find `if(row_reduction.empty())` more natural (and type safe), as it doesn't use the old C bool-int conversion.